### PR TITLE
Allow disabling runtime environments in assignments

### PR DESCRIPTION
**Pull request by Teyras**

### CODE REVIEW

File: app/V1Module/presenters/AssignmentsPresenter.php [hunk 1]
```diff
@@ -178,6 +188,13 @@ public function actionUpdateDetail(string $id) {
       throw new InvalidArgumentException("No entry for localized texts given.");
     }
 
+    if ($this->getHttpRequest()->getHeader("content-type") === "application/json") {
```
Comment (Neloop) [inline app/V1Module/presenters/AssignmentsPresenter.php]: Why is this necessary here?

File: app/model/entity/Assignment.php [hunk 2]
```diff
@@ -14,6 +14,7 @@
 use DateTime;
 use Kdyby\Doctrine\MagicAccessors\MagicAccessors;
 use Gedmo\Mapping\Annotation as Gedmo;
+use Spatie\Regex\Helpers\Arr;
```
Comment (Neloop) [inline app/model/entity/Assignment.php]: Is this used somewhere?

File: app/model/entity/Assignment.php [hunk 3]
```diff
@@ -407,13 +424,20 @@ function ($key, RuntimeEnvironment $env) {
             })
         ],
         "runtimeEnvironments" => [
-          "upToDate" => $this->getRuntimeEnvironments()->count() === $this->getExercise()->getRuntimeEnvironments()->count()
-            && $this->getRuntimeEnvironments()->forAll(function ($key, RuntimeEnvironment $env) {
+          "upToDate" => $this->runtimeEnvironments->count() === $this->getExercise()->getRuntimeEnvironments()->count()
```
Comment (Neloop) [inline app/model/entity/Assignment.php]: Well... question is, if synchronization should be advertised on change of environment which is disabled in assignment. This is rather unimportant for the assignment author. But not much of a big problem...

Comment (krulis-martin) [reply app/model/entity/Assignment.php]: Synchronization should not be advertised if the changes are in a disabled environment only. But I agree that it is not big issue.

File: app/V1Module/presenters/AssignmentsPresenter.php (hunk 1 above)
Comment (Teyras) [reply app/V1Module/presenters/AssignmentsPresenter.php]: Because the meaning of empty `disabledRuntimeEnvironments` is ambiguous when multipart/form-data is used.

File: app/model/entity/Assignment.php (hunk 2 above)
Comment (Teyras) [reply app/model/entity/Assignment.php]: Nope.

File: app/model/entity/Assignment.php (hunk 3 above)
Comment (Teyras) [reply app/model/entity/Assignment.php]: If it's not a big issue, I would probably leave this as is for now. I feel like rephrasing the condition for enabled environments only won't be that easy.

File: app/V1Module/presenters/AssignmentsPresenter.php (hunk 1 above)
Comment (Neloop) [reply app/V1Module/presenters/AssignmentsPresenter.php]: But this is quite ugly... Can you at least make some helper method here on BasePresenter or something like that?

File: app/model/entity/Assignment.php (hunk 3 above)
Comment (Neloop) [reply app/model/entity/Assignment.php]: True would not be... Agree to postpone this

**Comment by Teyras** [comment]
From my point of view the jedi are evil.

**Comment by krulis-martin** [comment]
Then you are lost!
